python/GafferUI/CompoundEditor.py

```diff
@@ -186,6 +186,9 @@ def _removeDetachedPanel( self, panel ) :
 		panel.__titleChangedConnection = None
 		panel._applyVisibility()
 
+		assert( not panel.visible() )
```

Comment:
I'm curious as to what's the purpose of this `assert`?
What do you gain by having it raise in case the panel is visible, now that you are using the `keepUntilIdle()`?

Reply:
My goal with having `keepUntilIdle()` rather than `executeOnIdle( parent.removeChild() )` was that all the visible effects (both in terms of what is on screen and what is visible via the API) were the same as before. So I wanted to check that the window was hidden while it was in its zombie state, so the user couldn't interact with it. I could have checked manually once, but I left the assert in because the various permutations of 2 or more Qt versions and 2 Python versions have a habit of throwing up unexpected results. Like all `asserts()`, it should never raise in practice, but if it does, it is something I need to know about and fix.